inwebo/inwebo-authenticator/org.wso2.carbon.identity.authenticator/src/main/java/org/wso2/carbon/identity/authenticator/CheckPushResult.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    /**
+     * Set the client certificate to Default SSL Context
+     *
+     * @param certificateFile File containing certificate (PKCS12 format)
+     * @param certPassword    Password of certificate
+     * @throws Exception
+     */
+    public static SSLContext setHttpsClientCert(String certificateFile, String certPassword)
+            throws Exception {
+        if (certificateFile == null || !new File(certificateFile).exists()) {
+            return null;
+        }
+        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+        KeyStore keyStore = KeyStore.getInstance("PKCS12");
+
+        InputStream keyInput = new FileInputStream(certificateFile);
+        keyStore.load(keyInput, certPassword.toCharArray());
+
+        keyInput.close();
+        keyManagerFactory.init(keyStore, certPassword.toCharArray());
+
+        SSLContext context = SSLContext.getInstance("TLS");
+        context.init(keyManagerFactory.getKeyManagers(), null, new SecureRandom());
+        SSLContext.setDefault(context);
+        return context;
+    }
+
+    /**
+     * Prompt for a login and an OTP and check if they are OK.
+     */
+    public JSONObject checkPushResult(String login, String sessionId) {
+        String urlParameters = null;
+        JSONObject json = null;
+        try {
+            urlParameters = "action=checkPushResult"
+                    + "&serviceId=" + URLEncoder.encode("" + serviceId, "UTF-8")
+                    + "&userId=" + URLEncoder.encode(login, "UTF-8")
+                    + "&sessionId=" + URLEncoder.encode(sessionId, "UTF-8")
+                    + "&format=json";
+        } catch (UnsupportedEncodingException e1) {
+            e1.printStackTrace();
+            json.put("err", "NOK:urlParams");
+        }
+        try {
+            if (this.context == null) {
+                try {
+                    this.context = setHttpsClientCert(this.p12file, this.p12password);
+                } catch (Exception e) {
+                    log.error("Error while adding the certificate" + e.getMessage(), e);
+                }
+            }
+            SSLSocketFactory sslsocketfactory = context.getSocketFactory();
+            URL url = null;
+            try {
+                url = new URL(urlString + urlParameters);
+            } catch (MalformedURLException e) {
+                log.error("Error while creating the URL" + e.getMessage(), e);
+            }
+            HttpsURLConnection conn = null;
+            try {
+                if (url != null) {
+                    conn = (HttpsURLConnection) url.openConnection();
+                }
+            } catch (IOException e) {
+                log.error("Error while creating the connection" + e.getMessage(), e);
+            }
+            if (conn != null) {
+                conn.setSSLSocketFactory(sslsocketfactory);
+            }
+
+            try {
+                if (conn != null) {
```

Comment:
if (conn != null) {

this if condition is duplicated in several places. Try to minimize the duplicate codes